### PR TITLE
Expose TexSubImage2D bad dimensions bug

### PR DESCRIPTION
**Pull request by qjia7**
This change will expose bug http://crbug.com/733172.

### CODE REVIEW

**Comment by qjia7** [comment]
@kenrussell @zhenyao PTAL. Thanks.

File: sdk/tests/js/tests/texture-corner-case-videos.js [hunk 1]
```diff
@@ -187,6 +187,20 @@ function generateTest(desc,
...
+                if (bindingTarget == gl.TEXTURE_CUBE_MAP) {
+                    continue;
+                }
+                gl.texSubImage2D(targets[tt], 0, 0, 0,
+                                 sourceSubRectangle[2], sourceSubRectangle[3],
+                                 gl[pixelFormat], gl[pixelType], videoElement);
+            } else {
+                gl.texSubImage2D(targets[tt], 0, 0, 0, gl[pixelFormat], gl[pixelType], videoElement);
+            }
+            wtu.glErrorShouldBe(gl, gl.NO_ERROR, "should be no errors");
```
Comment (zhenyao) [inline sdk/tests/js/tests/texture-corner-case-videos.js]: I am curious: if we add this error checking right after line 129, will it catch this bug? If yes, that seems a better place for this.

Comment (qjia7) [inline sdk/tests/js/tests/texture-corner-case-videos.js]: No, it won't catch this bug. The error is because we use line 152  'gl.texImage2D(targets[tt], 0, gl[internalFormat], gl[pixelFormat], gl[pixelType], videoElement)' to initialize the texture. In chromium, that API will go gpu-gpu path which will upload the video visible size data. But TexSubImage will call GLContext()->TexSubImage which use video nature size.   But this video has natural size > visible size. So when call texture->ValidForTexture in TextureManager::ValidateTexSubImage, it will not meet the condition  max_x <= info.width and print bad dimentions.

line 128 TexImage2D and TexSubImage2D both use nature size. So it can't catch this error.

Comment (zhenyao) [inline sdk/tests/js/tests/texture-corner-case-videos.js]: Thanks.  Then the current patch LGTM
